prowler/providers/aws/services/dms/dms_service.py

```diff
@@ -13,7 +15,11 @@ def __init__(self, provider):
         self.instances = []
         self.endpoints = {}
         self.__threading_call__(self._describe_replication_instances)
+        self.__threading_call__(
+            self._list_tags, [instance.arn for instance in self.instances]
+        )
         self.__threading_call__(self._describe_endpoints)
+        self.__threading_call__(self._list_tags, list(self.endpoints.keys()))
```

Comment:
Please, only call the `list_tags` function once and iterate through the necessary resources inside the function.